rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -120,18 +153,62 @@ trait NodeInfoService {
    * @return
    */
   def getAllSystemNodeIds() : Box[Seq[NodeId]]
+
+
+  /**
+   * Getting something like a nodeinfo for pending / deleted nodes
+   */
+  def getPendingNodeInfos(): Box[Map[NodeId, NodeInfo]]
```

Comment:
AS saud above I see how, but it's complicated and probably risky (but it would provide a cache for deleted/pending nodes, not sure we want it though ... )